### ISSUE TITLE
Tighten validation for Bank Referral Scheme

### ISSUE THREAD

**Issue by raoulsullivan** [FundingOptions/sme-finance-application-schema#16]
The schema are currently lacking validation required for the Bank Referral Scheme. Improving this will be necessary prior to scheme launch, but will be a change which breaks/limits use of the schema in a non-referral context.

**Comment by raoulsullivan** [comment]
https://www.fundingoptions.com/schemas/#compliance

**Comment by raoulsullivan** [comment]
Olivier suggests that we communicate the required, recommended and optional information from sme_vX by splitting sme_vX into components to this effect. This would have two advantages:

1) We can communicate 'recommended', which is not possible under json-schema, ensuring these schema does not need additional information to be able to be used effectively
2) Improves readability by making the important bits (required/recommended) less intimidating than sme_v4

Question to @nesaro - I can think of two ways of doing this, can you advise/improve? Either:

1) Add the relevant properties directly to sme_finance_application_vX and remove the sme_vX property:

```
finance_application:{
    sme_required:{},
    sme_recommended:{},
    sme_optional:{}
}
```

Or

2) Turn sme_vX into a wrapper with 3 properties:

```
finance_application: {
    sme: {
        required: {},
        recommended:{},
        optional:{}
    }
}
```

**Comment by raoulsullivan** [comment]
Commentary by @nesaro pending - 2 alternative approaches to communicating this information

**Comment by raoulsullivan** [comment]
In conversation by email, myself, Olivier (FundingXchange), Nestor (Funding Options) and Mitul (BizFiTech) covered 4 different approaches to the problem of communicating 'required', 'recommended' and 'extra' information.

4 possible solutions were discussed:

1. Use a hierarchical design of nested properties (as proposed by Olivier)
2. Use lists (as proposed by Nestor)
3. Use properties (rejected by myself and Nestor as too messy to implement)
4. Use 'required' and description (as proposed by myself)

In the end consensus was reached and we implemented solution 4 in pull request #25

**Comment by raoulsullivan** [comment]
Validation changes from HMT in #34 

The issue with date_finance_requested vs date_finance_required is also unresolved
